### PR TITLE
Intsights mssp feature

### PR DESCRIPTION
**Pull request by funerr**
## Status
Ready

## Related Issues
fixes: [https://github.com/demisto/etc/issues/14894](https://github.com/demisto/etc/issues/14894)

## Description
Adds the ability to work on MSSP (sub) accounts via the Intsights Mssp Id parameter.
Basically, if you are a MT account in insights, you can now use the Intsights integration on a sub-account (in the intsight system).

## Screenshots
image

## Related PRs

branch | PR
------ | ------

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
   - No (not known)

## Must have
- [X] Tests
- [X] Documentation (with link to it)
-- [https://github.com/demisto/etc/issues/15394](https://github.com/demisto/etc/issues/15394)
- [x] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.

## Additional changes
Added a new test as we need a new instance to check the MT features.

## Notes:
Tests may fail until: [https://github.com/demisto/content-test-conf/pull/315](https://github.com/demisto/content-test-conf/pull/315) is merged.

### CODE REVIEW

File: Integrations/integration-IntSights.yml [hunk 1]
```diff
@@ -832,10 +844,76 @@ script:
         })
 
 
+    def get_mssp_sub_accounts():
+        account_id = demisto.getParam('credentials')['identifier']
+        accounts = req('GET', 'public/v1/mssp/customers').json()
+        demisto.results({
+            'Type': entryTypes['note'],
+            'EntryContext': {'IntSights.MsspAccounts': accounts},
```
Comment (liorblob) [inline Integrations/integration-IntSights.yml]: Should Be `IntSight.MsspAccount`, also need to use DT for unique entries (e.g., `IntSight.MsspAccount(val.ID === obj.ID)`)

File: Integrations/integration-IntSights.yml [hunk 2]
```diff
@@ -832,10 +844,76 @@ script:
...
+            'HumanReadable': tableToMarkdown('IntSights MSSP accounts for ' + account_id, [a for a in accounts], ["_id", 'CompanyName', "Status"]),
```
Comment (liorblob) [inline Integrations/integration-IntSights.yml]: Maybe map the accounts array for human readable headers (`_id` -> `ID`)
also make sure you use single quote with strings.

File: Integrations/integration-IntSights.yml [hunk 3]
```diff
@@ -832,10 +844,76 @@ script:
...
+    def test_mssp():
+        account_id = demisto.getParam('credentials')['identifier']
+
+        # Check if mssp_sub_account_id is full or not
+        if not mssp_account_id:
+            return_error('Please insert an mssp sub account id, before preforming this action.')
+
+        # Check if account exsits
+        HEADERS['Account-Id'] =  account_id
+        accounts = req('GET', 'public/v1/mssp/customers').json()
```
Comment (liorblob) [inline Integrations/integration-IntSights.yml]: Make sure it's a valid json

File: Integrations/integration-IntSights.yml [hunk 4]
```diff
@@ -832,10 +844,76 @@ script:
...
+        account_2 = account_ids[1]
+        HEADERS['Account-Id'] =  account_2
+        account_2_ua = req('GET', 'public/v1/account/used-assets').text
+
+        if account_1_ua == account_2_ua:
+            demisto.log("[DEBUG] - MSSP Sub account results ({}={})".format(str(account_1_ua), str(account_2_ua)))
+            return_error("MSSP test failed, sub accounts are not different.")
+
+        accounts[0].update(json.loads(account_1_ua))
+        # demisto.log(str(accounts))
```
Comment (liorblob) [inline Integrations/integration-IntSights.yml]: Remove

File: Integrations/integration-IntSights.yml [hunk 5]
```diff
@@ -832,10 +844,76 @@ script:
...
+        # Call account 1 with /public/v1/account/used-assets after update of account-id in the header
+        account_1 = account_ids[0]
+        HEADERS['Account-Id'] =  account_1
+        account_1_ua = req('GET', 'public/v1/account/used-assets').text
+
+        # Call account 2 and check the difference
+        account_2 = account_ids[1]
+        HEADERS['Account-Id'] =  account_2
+        account_2_ua = req('GET', 'public/v1/account/used-assets').text
+
```
Comment (liorblob) [inline Integrations/integration-IntSights.yml]: Validate responses

File: Integrations/integration-IntSights.yml [hunk 6]
```diff
@@ -832,10 +844,76 @@ script:
...
+        accounts[0].update(json.loads(account_1_ua))
+        # demisto.log(str(accounts))
+        accounts[1].update(json.loads(account_2_ua))
+
+        demisto.results({
+            'Type': entryTypes['note'],
+            'EntryContext': {'IntSights.MsspAccounts': accounts},
+            'HumanReadable': tableToMarkdown('IntSights MSSP accounts used assets ' + account_id, [a for a in accounts], \
+                ["_id", 'CompanyName', "Status", "AssetsLimit", "AssetsCount"]),
+            'Contents': accounts,
```
Comment (liorblob) [inline Integrations/integration-IntSights.yml]: Same as above

File: Integrations/integration-IntSights.yml [hunk 7]
```diff
@@ -1584,7 +1663,15 @@ script:
       description: The alert's closed reason
       type: string
     description: Close an alert
+  - name: intsights-mssp-get-sub-accounts
+    arguments: []
+    description: Get all mssp sub accounts
+  - name: intsights-mssp-test
+    arguments: []
+    description: Tests intsights mssp features, should be an mssp account.
```
Comment (liorblob) [inline Integrations/integration-IntSights.yml]: Need to add outputs - `IntSights.MsspAccount.ID`, `IntSights.MsspAccount.Status`, etc.

File: Tests/id_set.json [hunk 8]
```diff
@@ -12757,7 +12760,7 @@
                     "DeleteContext"
                 ], 
                 "command_to_integration": {
-                    "whois": "whois"
+                    "whois": "Whois"
```
Comment (liorblob) [inline Tests/id_set.json]: Whois? why is this changed?

File: Integrations/integration-IntSights.yml [hunk 9]
```diff
@@ -45,6 +45,11 @@ configuration:
   defaultvalue: ""
   type: 13
   required: false
+- display: '(For MSSP accounts only) Sub account ID:'
```
[the remark this answers is not in the document]
Comment (anara123) [reply Integrations/integration-IntSights.yml]: I think it is necessary. Makes things more clear

File: Integrations/integration-IntSights.yml (hunk 1 above)
Comment (funerr) [reply Integrations/integration-IntSights.yml]: @liorblob, where can I read more about this notation? I don't really understand the concept.

File: Integrations/integration-IntSights.yml (hunk 2 above)
Comment (funerr) [reply Integrations/integration-IntSights.yml]: I didn't want to do that because it is information straight from IntSights. So, Displaying their conventions makes it more clear as to where the information is from (how can you know the ID is not a Demisto ID?)

Comment (liorblob) [reply Integrations/integration-IntSights.yml]: Is it displayed like this in the IntSights UI? Anyway, for context there is also a description for each output. It just looks better this way IMO.

File: Tests/id_set.json (hunk 8 above)
Comment (funerr) [reply Tests/id_set.json]: Idk, I just ran the automated script. Should I change it back?

Comment (liorblob) [reply Tests/id_set.json]: Compare it with master and take what's in there

Comment (funerr) [reply Tests/id_set.json]: It's the correct one...
